tasktiger/executor.py

```diff
@@ -80,11 +80,88 @@ def execute(
             locks: List of task locks to renew periodically.
             queue_lock: Optional queue lock to renew periodically for max
                 workers per queue.
+
+        Returns:
+            Whether task execution was successful.
         """
         raise NotImplementedError
 
+    def execute_tasks(self, tasks: List[Task], log: BoundLogger) -> bool:
+        """
+        Executes the tasks in the current process. Multiple tasks can be passed
+        for batch processing. However, they must all use the same function and
+        will share the execution entry.
+        """
+        success = False
+
+        execution: Dict[str, Any] = {}
+
+        assert len(tasks)
+        task_func = tasks[0].serialized_func
+        assert all([task_func == task.serialized_func for task in tasks[1:]])
+
+        execution["time_started"] = time.time()
+
+        try:
+            func = tasks[0].func
+
+            runner_class = get_runner_class(log, tasks)
+            runner = runner_class(self.tiger)
+
+            is_batch_func = getattr(func, "_task_batch", False)
+            g["tiger"] = self.tiger
```

Comment:
Would it make sense to unset that (and other parts of `g`) at the end of the execution?